protobufs/livekit_sip.proto

```diff
@@ -199,10 +209,34 @@ message SIPInboundTrunkInfo {
   bool krisp_enabled = 13;
 }
 
+message SIPInboundTrunkUpdate {
```

Comment:
To benjamin's point, I think it's simpler to allow a user to set the entire field together.. vs add/remove semantics.

i.e. wdyt about `repeated string numbers; repeated string allowed_addresses, etc`?

Reply:
Done, removed these set/add/del fields.

However, having only set field leads to an issue: we cannot detect if it's set or not. Thus, we cannot really set it to empty list, which might be useful for `allowed_numbers` and `allowed_addresses`.

For now I added a special valued of `[""]` that is interpreted as "set to empty list". But it's just a hack. A better way could be to add a separate message like `message ListUpdate { repeated string arr = 1; }` and use it as a type of the field: `ListUpdate allowed_numbers = x;`. It also allows us to go back and implement add/del in the future under the same message.

Reply:
fwiw, for project updates, we have simple "update if not nil" semantics for most fields, with upsert/delete semantics for maps. Not sure if that would be applicable here: https://github.com/livekit/backend-common/blob/c78e4b02dd1facf497baec49c9ac651053c77f9e/project.proto#L92

Reply:
yeah repeat types are a bit harder to reset.. @dennwc do you prefer to do this or adding a field for `optional bool clear_allowed_numbers` ?

Reply:
I'll use the `ListUpdate` approach then. `clear_allowed_numbers` is easier to accidentally miss.

Reply:
As for the attribute/headers, I'll use the same approach as the SDK uses for attributes: upsert everything and deleted keys with zero value.